File: src/main/java/priv/bajdcc/LALR1/semantic/tracker/TrackerError.java
package priv.bajdcc.LALR1.semantic.tracker;

import priv.bajdcc.util.Position;

/**
 * 错误
 *
 * @author bajdcc
 */
public class TrackerError {

	/**
	 * 错误信息
	 */
	public String message = "";

	/**
	 * 位置
	 */
	public Position position = null;

	public TrackerError(Position pos) {
		position = pos;
	}

	@Override
	public String toString() {
		return String.format("位置：[%s] 信息：%s", position.toString(),
				message);
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/semantic/tracker/TrackerError.kt
package com.bajdcc.LALR1.semantic.tracker

import com.bajdcc.util.Position

/**
 * 错误
 * [position] 位置
 * [message] 错误信息
 * @author bajdcc
 */
data class TrackerError(var position: Position, var message: String = "") {

    override fun toString(): String {
        return "位置：[$position] 信息：$message"
    }
}
